masim.c

```diff
@@ -14,6 +14,30 @@
 #include "misc.h"
 #include "masim.h"
 
+// hugepage definitions
+#define PROTECTION (PROT_READ | PROT_WRITE)
+
+#ifndef MAP_HUGETLB
+#define MAP_HUGETLB 0x40000 /* arch specific */
```

Comment:
Why `MAP_HUGETLB` wouldn't defined?  And, I thought it's not arch-specific?

```
linux$ git grep "define MAP_HUGETLB" include/
include/uapi/asm-generic/mman-common.h:#define MAP_HUGETLB              0x040000        /* create a huge page mapping */
```

Reply:
yes right, I missed this